docs/Agent/user-guide/guides/end-to-end-workflow.md

```diff
@@ -0,0 +1,21 @@
+### End-to-End Workflow
+
+Each command in pbench-agent accepts the `--help` option and outputs a brief usage message
+
+The default set of tools for data collection can be enabled with
+
+	$pbench-register-tool-set
+
+To list all your registered tools
+
+	$pbench-list-tools
+
+You may then perform a built-in benchmark by running it's Pbench script
+
+	$pbench-user-benchmark – sleep 10
```

Comment:
I'm not sure this is a great example without a real test script. It's also not a good example of the sentence that introduces it: we're not "perform[ing] a built-in benchmark"; we're *sleeping* with the generic wrapper to gather performance data. A "built-in benchmark" would be uperf, fio, etc.

(Grammatically, "it's" is used as a contraction for "it is" or "it has", or as a possessive form only for *people*, not for *things*; so in this sentence you should properly use "its" ... or maybe even just "the associated Pbench script".)

Finally, we need *two* hyphens to stop option processing on the `pbench-user-benchmark` command to introduce the "benchmark" script (`sleep 10` in this case). I only see one, although it looks oddly like an em-dash here. (The reference above to `--help` clearly shows two distinct hyphens, which is what we need here.)